hmpps-kotlin-spring-boot-autoconfigure/src/main/kotlin/uk/gov/justice/hmpps/kotlin/auth/HmppsWebClientConfiguration.kt

```diff
@@ -45,20 +45,39 @@ private const val DEFAULT_HEALTH_TIMEOUT_SECONDS: Long = 2
 @ConditionalOnBean(ClientRegistrationRepository::class)
 @Configuration
 class HmppsWebClientConfiguration {
+
+  /**
+   * This method generates an instance of the [AuthorizedClientServiceOAuth2AuthorizedClientManager]
+   * class configured to cache all OAuth2 tokens under a single **principalName** using the
+   * [GlobalPrincipalOAuth2AuthorizedClientService].
+   *
+   * The purpose of this [OAuth2AuthorizedClientManager] is to avoid unnecessary token requests to HMPPS Auth,
+   * and it should be used for web clients where a username is being injected to the token request.
```

Comment:
I guess this solution isn't required where there is no username because the built in Spring cache by principal will work? But people will get this solution by default anyway? Does that matter?

Reply:
I have reworded this now so hopefully it makes more sense. Essentially the `GlobalPrincipalOAuth2AuthorizedClientService` is the default however if you want to configure a webclient to include the username parameter in the token request this requires a different `OAuth2AuthorizedClientService` as the token will need to be cached on a per user (principal) basis. I have also added a note on this in the docs.